README.md
# Creating a prerelease

```
$ git checkout master
$ npm version prerelease --preid=beta
$ npm run publish
```

# Creating a release

```
$ git checkout master
$ npm version [major|minor|patch]
$ npm run publish


History.md

1.6.0 / 2019-10-23
==================

  * JSON parse improvements and rename finboxio to finbox
  * Update branding & Finbox urls, fix utf encoding issue
  * Upgrade/installation bug fixes

1.5.0 / 2019-07-05
==================

  * Remove S&P toggle
  * Patch Mac installation location


1.4.2-beta.0 / 2019-07-05
=========================

  * Remove S&P toggle and patch Mac installation location

1.4.1 / 2019-06-03
==================

  * Release 1.4.1

1.4.1-beta.0 / 2019-06-03
=========================

  * Reverse logic for determining if system libc needs to be used

1.4.0 / 2019-05-09
==================

  * Change default api version to v3 data
  * Update to support 16.16.9 on OSX

1.3.2-beta.2 / 2019-05-09
=========================

  * Update button text for reverting to legacy api

1.3.2-beta.1 / 2019-05-09
=========================

  * Update button text for reverting to legacy api

1.3.2-beta.0 / 2019-05-09
=========================

  * Change default api version to v3 data
  * Update to support 16.16.9 on OSX

1.3.1 / 2019-03-14
==================

  * Fix clib support for new release 16.16.8

1.3.0 / 2019-02-18
==================

  * Fix for compatibility with Excel for Mac v16.16.7
  * Update beta icons for mac compatibility
  * Add option to toggle beta api

1.3.0-beta.1 / 2019-02-15
=========================

  * Fix for compatibility with Excel for Mac v16.16.7

1.3.0-beta.0 / 2019-02-13
=========================

  * Update beta icons for mac compatibility

1.2.1-beta.0 / 2019-02-13
=========================

  * Add option to toggle beta api

1.2.0 / 2019-01-19
==================

  * Fix an integer overflow error for users with increased quota
  * Make null value configurable and default to 0
  * Send structured client header with system info in X-Finboxio-Addon
  * Adds configuration option for finboxioApiUrl
  * Update for compatibility with Excel for Mac v16.21+
  * Fix Mac installation bug

1.1.7-beta.3 / 2019-01-19
=========================

  * Include Mac v16.21 in the compatibility fix

1.1.7-beta.2 / 2019-01-19
=========================

  * Fix integer overflow error with increased quota

1.1.7-beta.1 / 2019-01-18
=========================

  * Add workbook name to API header
  * Make default null value configurable
  * Send structured client header in X-Finboxio-Addin
  * Make api url configurable

1.1.7-beta.0 / 2019-01-18
=========================

  * Update for compatibility with Excel for Mac v16.22
  * Fix Mac installation bug

1.1.6 / 2018-11-17
==================

  * Fixes bug with Mac update that broke VBA.MkDir

1.1.5 / 2018-11-01
==================

  * Update README

1.1.5-beta.1 / 2018-11-01
=========================

  * 1.1.5-beta.0
  * Try with beta prefix

1.1.5-beta.0 / 2018-11-01
=========================

  * Try with beta prefix

1.1.5-4 / 2018-11-01
====================

  * Fix release-notes one more time

1.1.5-3 / 2018-11-01
====================

  * Fix release notes script

1.1.5-2 / 2018-11-01
====================

  * Fix release-notes script

1.1.5-1 / 2018-11-01
====================

  * Add readme with release instructions

1.1.5-0 / 2018-11-01
====================

  * reset version

1.1.5-beta.0 / 2018-11-01
=========================

  * reset version

1.1.4-beta0.0 / 2018-10-29
==========================

  * Prerelease 1.1.4-beta0
  * Merge pull request #9 from finboxio/fix/vba-install-bug
  * Fix installation bug with new version of VBA

1.1.4 / 2018-01-23
==================

  * Support Mac 2016 v16.*

1.1.4-2 / 2018-01-22
====================

  * Mac v16.* compatibility

1.1.4-1 / 2018-01-22
====================

  * safer publish script

1.1.4-0 / 2018-01-22
====================

  * safer publish script

1.1.3-0 / 2018-01-22
====================

  * Support Mac 2016 v16.*

1.1.2 / 2017-09-14
==================

  * Fixes layout of Mac login warning

1.1.1 / 2017-09-14
==================

  * Add warning to Mac2016 login prompt about missing cursor

1.1.0 / 2017-09-14
==================

  * Switches default parse method on windows to iterate to improve performance for large workbooks
  * Adds settings module to functions add-in
  * Adds settings module to functions add-in

1.0.0 / 2017-09-12
==================

  * Improved performance with key batching
  * Better support for nested and table-based formulas
  * Support for international date and number formats
  * Improved installation flow
  * Fully automatic upgrades
  * Login persistence
  * Real-time quota indicator
  * Many other improvements related to stability and maintainability

0.24.5 / 2017-09-12
===================

  * Completed install/upgrade tests for Win2013
  * Fixes link replacement for links with unquoted filenames
  * Completed install/upgrade tests for Win2007 & Win2010
  * Completed install/upgrade tests for Mac2016
  * Finished Mac2011 install/upgrade tests

0.24.4 / 2017-09-12
===================

  * Fix installation cancel on Mac2011

0.24.3 / 2017-09-12
===================

  * Removes debug statement from uninstall procedure

0.24.2 / 2017-09-12
===================

  * Fixes uninstall on Mac2011

0.24.1 / 2017-09-12
===================

  * Improvements to FNBX link-replacer

0.24.0 / 2017-09-12
===================

  * Minor version bump

0.23.8 / 2017-09-12
===================

  * Adds FNBX help context for Windows users

0.23.7 / 2017-09-12
===================

  * Reload finboxio menu completely when opening add-in functions

0.23.6 / 2017-09-12
===================

  * Adds cleanup logic for older Mac2016 add-in versions installed in the wrong place

0.23.5 / 2017-09-11
===================

  * Ensures finboxio.xlam is fully removed before installing
  * Closes install dialog on Mac2016

0.23.4 / 2017-09-11
===================

  * Promote prerelease

0.23.4-0 / 2017-09-11
=====================

  * Fixes download module for Excel 2007

0.23.3 / 2017-09-11
===================


src/finbox.install.xlam/ConfigModule.bas
Attribute VB_Name = "ConfigModule"
Option Explicit
Option Private Module

Public Const RELEASES_URL = "https://api.github.com/repos/finboxio/xlam/releases"
Public Const DOWNLOADS_URL = "https://github.com/finboxio/xlam/releases/download"

Public Const AddInInstalledFile = "finbox.xlam"
Public Const LegacyInstalledFile = "finboxio.xlam"
Public Const AddInInstallerFile = "finbox.install.xlam"
Public Const AddInFunctionsFile = "finbox.functions.xlam"
Public Const LegacyFunctionsFile = "finboxio.functions.xlam"
Public Const AddInKeyFile = "finbox.key"
Public Const AddInSettingsFile = "finbox.cfg"
Public Const AddInLogFile = "finbox.log"

' These will be loaded on Workbook_Open
Public AddInInstalled As Boolean
Public cwd As String

Public Function AddInManagerFile() As String
    AddInManagerFile = ThisWorkbook.name
End Function

Public Function StagingFile(file As String) As String
    StagingFile = VBA.Left(file, VBA.InStrRev(file, ".")) & "staged" & VBA.Mid(file, InStrRev(file, "."))
End Function

Public Function StagingPath(file As String) As String
    StagingPath = LocalPath(StagingFile(file))
End Function

Public Sub cd(path As String)
    If VBA.Right(path, 1) = Application.PathSeparator Then
        cwd = VBA.Left(path, VBA.Len(path) - 1)
    Else
        cwd = path
    End If
End Sub

Public Function LocalPath(file As String) As String
    If cwd = "" Then cwd = ThisWorkbook.path
    LocalPath = cwd & Application.PathSeparator & file
End Function

Public Function AddInVersion(Optional file As String) As String
    If file = "" Then file = ThisWorkbook.name
    On Error Resume Next
    AddInVersion = Workbooks(file).Sheets("finbox").Range("AppVersion").value
    AddInVersion = Workbooks(file).Sheets("finboxio").Range("AppVersion").value
End Function

Public Function AddInReleaseDate(Optional file As String) As Date
    If file = "" Then file = ThisWorkbook.name
    On Error Resume Next
    AddInReleaseDate = VBA.Now()
    AddInReleaseDate = Workbooks(file).Sheets("finbox").Range("ReleaseDate").value
    AddInReleaseDate = Workbooks(file).Sheets("finboxio").Range("ReleaseDate").value
End Function

Public Function AddInLocation(Optional file As String) As String
    If file = "" Then file = ThisWorkbook.name
    On Error Resume Next
    AddInLocation = Workbooks(file).FullName
End Function

Public Function SafeDir(file As String, Optional attributes As VbFileAttribute) As String
    On Error Resume Next
    SafeDir = VBA.dir(file, attributes)
End Function

Public Sub SafeMkDir(path As String)
    Dim folder As String
    folder = path
    If Right(path, 1) = Application.PathSeparator Then
        folder = Left(path, Len(path) - 1)
    End If
    If SafeDir(folder, vbDirectory) = vbNullString Then
        #If Mac Then
            Dim appleScript As String
            appleScript = "do shell script ""mkdir -p '" & folder & "'"""
            MacScript appleScript
        #Else
            VBA.MkDir folder
        #End If
    End If
End Sub

Public Function ExcelVersion() As String
    Dim version As Integer: version = MSOfficeVersion
    ExcelVersion = "Unsupported"
    
    #If Mac Then
        If version = 14 Then
            ExcelVersion = "Mac2011"
        ElseIf version = 15 Then
            ExcelVersion = "Mac2016"
        ElseIf version = 16 Then
            ExcelVersion = "Mac2016"
        End If
    #Else
        If version = 12 Then
            ExcelVersion = "Win2007"
        ElseIf version = 14 Then
            ExcelVersion = "Win2010"
        ElseIf version = 15 Then
            ExcelVersion = "Win2013"
        ElseIf version = 16 Then
            ExcelVersion = "Win2016"
        End If
    #End If
End Function

' Returns the version of MS Office being run
'    9 = Office 2000
'   10 = Office XP / 2002
'   11 = Office 2003 & LibreOffice 3.5.2
'   12 = Office 2007
'   14 = Office 2010 or Office 2011 for Mac
'   15 = Office 2013 or Office 2016 for Mac
'   16 = Office 2016 for Mac or Windows
Public Function MSOfficeVersion() As Integer
    Dim verStr As String
    Dim startPos As Integer
    MSOfficeVersion = 0
    verStr = Application.version
    startPos = VBA.InStr(verStr, ".")
    On Error Resume Next
    If startPos > 0 Then
        MSOfficeVersion = CInt(VBA.Left(verStr, startPos - 1))
    Else
        MSOfficeVersion = CInt(verStr)
    End If
End Function

Sub auto_add()
End Sub
Sub auto_remove()
End Sub
